File: Selenium/Project/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# URL of the website
url = "https://alchemy.hguy.co/lms/"

# Initialize Firefox WebDriver (make sure you have geckodriver in your PATH)
driver = webdriver.Firefox()

# Open the website
driver.get(url)
# Print the title of the page
print("Page title is : ",driver.title)

# Wait for the Contact tab to be clickable
contact_tab = WebDriverWait(driver, 10).until(
    EC.element_to_be_clickable((By.LINK_TEXT, "Contact"))
)

# Click on the Contact tab
contact_tab.click()

# Wait for the Full Name and Email fields to be present
full_name_field = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "wpforms-8-field_0"))
    )
email_field = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "wpforms-8-field_1"))
    )
subject_field = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "wpforms-8-field_3"))
    )
comment_field = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "wpforms-8-field_2"))
    )

    
    # Enter details in the Full Name and Email fields
full_name_field.send_keys("Nitish Srivastava")
email_field.send_keys("nitish@sky.com")
subject_field.send_keys("Python")
comment_field.send_keys("Hello World")

send_message = WebDriverWait(driver, 10).until(
        EC.presence_of_element_located((By.ID, "wpforms-submit-8"))
    )
send_message.click()
message = driver.find_element(By.ID, "wpforms-confirmation-8")
print("Submission message: ", message.text)

# Close the browser window
driver.quit()

File: Selenium/Project/Activity8.java
package project;

import org.testng.annotations.Test;

import io.github.bonigarcia.wdm.WebDriverManager;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;


public class Activity8 {

	WebDriver driver;

	@BeforeClass
	public void beforeClass() {
		WebDriverManager.firefoxdriver().setup();
		driver = new FirefoxDriver();

		// Open browser
		driver.get("https://alchemy.hguy.co/lms/");

	}

	@Test
	public void Contact_us_and_complete_form() {
		driver.findElement(By.xpath("//*[contains(text(),'Contact')]")).click();
		String Page_title = driver.getTitle();
		System.out.println("Title of the Page is : " + Page_title);
		driver.findElement(By.id("wpforms-8-field_0")).sendKeys("Nitish Srivastava");
		driver.findElement(By.id("wpforms-8-field_1")).sendKeys("nitish@sky.com");
		driver.findElement(By.id("wpforms-8-field_3")).sendKeys("TestNG");
		driver.findElement(By.id("wpforms-8-field_2")).sendKeys("Hello World!!");
		driver.findElement(By.id("wpforms-submit-8")).click();
		String Display_message = driver.findElement(By.id("wpforms-confirmation-8")).getText();
		System.out.println("Message after submission : " + Display_message);
	}

	@AfterClass
	public void afterClass() {
		driver.close();
	}

}